src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -1652,6 +1656,31 @@ private static Boolean isVideoMetaMediaType(ExtBidPrebidMeta meta) {
                 .orElse(false);
     }
 
+    private static Integer resolveAdvertiserId(ObjectNode bidExt) {
+        return Optional.ofNullable(bidExt)
+                .map(ext -> ext.get("rp"))
+                .filter(JsonNode::isObject)
+                .map(rp -> rp.get("advid"))
+                .map(RubiconBidder::convertToInt)
+                .orElse(null);
+    }
+
+    private static Integer convertToInt(JsonNode jsonNode) {
+        if (jsonNode.canConvertToInt()) {
+            return jsonNode.asInt();
+        }
+
+        if (jsonNode.isTextual()) {
+            try {
+                return Integer.parseInt(jsonNode.asText());
```

Comment:
Using exceptions for control flow can burn quite a lot of CPU power, due to exceptions being created and immediately thrown away.

I would recommend first checking if the text node can be converted to an `Int` and then parse it. Feel free to read the blog post I wrote, when I got hit very hard by this 😅  https://medium.com/@muuki88/follow-the-stacktraces-jvm-performance-profiling-3c371d323e5f